includes/search.form.inc

```diff
@@ -50,7 +50,9 @@ function islandora_collection_search_form($form, &$form_state) {
       module_load_include('inc', 'islandora_basic_collection', 'includes/utilities');
       $child_object = islandora_object_load($collection);
       $parents = islandora_basic_collection_get_parent_pids($child_object);
-      $collections = drupal_map_assoc($parents);
+      $collections = array_filter(drupal_map_assoc($parents), function ($parent) {
+        return islandora_object_load($parent);
```

Comment:
Alternatively:
```php
$collections = array_filter(drupal_map_assoc($parents, 'islandora_object_load'));
```

... And then the `islandora_object_load()` calls below could be squashed.